### PR TITLE
ENH: using dpnp/dpctl instead of numpy in onedal4py

### PR DESCRIPTION
**Pull request by samir-nasibli**
# Description
[WIP]
Using `dpnp`/`dpctl` instead of `numpy` in onedal4py:
* using `sua_iface` for conversions

# TODO
* address TODOs
* dpnp into CI
* dispatcher on sklearnex level.

### CODE REVIEW

File: onedal/basic_statistics/basic_statistics.py [hunk 1]
```diff
@@ -17,8 +17,11 @@
 from sklearn.base import BaseEstimator
 from abc import ABCMeta, abstractmethod
 
-import numpy as np
-from numbers import Number
+# TODO:
+# this interface will be wrapped in _data_conversion module.
+import dpctl.tensor as dpt
+
+import dpnp
```
Comment (KulikovNikita) [inline onedal/basic_statistics/basic_statistics.py:24]: What about just importing `dpnp` as `np`?

Comment (samir-nasibli) [reply onedal/basic_statistics/basic_statistics.py:24]: 1) dpnp doesn't support all numpy functionality.

2) This is draft code. I will moved explicit data frame call into separate module

**Comment by samir-nasibli** [comment]
This PR is just for some testing were created. Will be closed.

**Comment by samir-nasibli** [comment]
some of dev on #1397

**Pull request closed without merging**